rooms/dojo.py

```diff
@@ -0,0 +1,345 @@
+                    staff = self.add_staff(name)
+                    return staff
+            else:
+                print("Person cannot be created due to invalid designation!")
+        else:
+            print("Person cannot be created with an empty name!")
+
+    def add_fellow(self, name, accommodation):
+        """
+        This function create a fellow and add it to the list of staff
+        while calling the allocate function to allocate room.
+        """
+        new_fellow = Fellow(name, "fellow")
+        new_fellow.generate_id("fellow", self.fellow_list)
+        new_fellow.office = self.allocate_room(new_fellow, Office)
+        if accommodation.upper() == "Y":
+            new_fellow.livingspace = self.allocate_room(new_fellow, \
+                                                            LivingSpace)
+            new_fellow.wants_accommodation = True
+        self.fellow_list.append(new_fellow)
+        return new_fellow
+
+    def add_staff(self, name):
+        """
+        This function create a staff and add it to the list of staff
+        while calling the allocate function to allocate room.
+        """
+        new_staff = Staff(name, "staff")
+        new_staff.generate_id("staff", self.staff_list)
+        new_staff.office = self.allocate_room(new_staff, Office)
+        self.staff_list.append(new_staff)
+        return new_staff
+
+    def check_room_name_exist(self, room_name):
+        """
+        This function checks if a room name passed already existed
+        in the list of all rooms.
+        """
+        return room_name in self.all_rooms
+
+    def get_available_rooms(self, room_type):
+        """
+        This function gets the list of availble rooms
+        of a specified room type
+        """
+        available_room = []
+        for room in self.all_rooms:
+            room_available = self.all_rooms[room].total_space > \
+                                    len(self.allocated[room]) \
+                                        if room in self.allocated else True
+
+            if room_available != False and \
+                        isinstance(self.all_rooms[room], room_type):
+                available_room.append(self.all_rooms[room])
+        return available_room
+
+
+    def allocate_room(self, person, room_type):
+        """
+        This function randomly assign room to a person
+        from a list of rooms that are available_room
+        """
+        available_rooms = self.get_available_rooms(room_type)
+        if len(available_rooms) > 0:
+            room = random.choice(available_rooms)
+            if not room.name in self.allocated:
+                self.allocated[room.name] = []
+            self.allocated[room.name].append(person)
+            return room
+        else:
+            if room_type == Office:
+                self.unallocated["office"].append(person)
+            else:
+                self.unallocated["livingspace"].append(person)
+            return None
```

Comment:
Thanks, I don't want to leave the function without returning anything but I will remove it since I discovered that python return false automatically when nothing is returned